fl4health/model_bases/autoencoders_base.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
+from typing import Optional, Tuple
+
+import torch
+import torch.nn as nn
+
+from fl4health.utils.dataset_converter import AutoEncoderDatasetConverter
+
+
+class AbstractAe(nn.Module, ABC):
+    """The base class for all Encoder-Decoder based models.
+    All we need to define such model is the type of the model, and the structure
+    of the encoder and the decoder modules. This type of model should have the capability
+    to encode data using the encoder module and decode the output of the encoder using the decoder module.
+    """
+
+    def __init__(
+        self,
+        encoder: nn.Module,
+        decoder: nn.Module,
+    ) -> None:
+        super().__init__()
+        self.encoder = encoder
+        self.decoder = decoder
+
+    @abstractmethod
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        # Forward is called in client classes with a single input tensor.
+        raise NotImplementedError
+
+
+class BasicAe(AbstractAe):
+    def __init__(self, encoder: nn.Module, decoder: nn.Module) -> None:
+        super().__init__(encoder, decoder)
+
+    def encode(self, input: torch.Tensor) -> torch.Tensor:
+        latent_vector = self.encoder(input)
+        return latent_vector
+
+    def decode(self, latent_vector: torch.Tensor) -> torch.Tensor:
+        output = self.decoder(latent_vector)
+        return output
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        z = self.encode(input)
+        return self.decode(z)
+
+
+class VariationalAe(AbstractAe):
+    """Variational Auto-Encoder model base class."""
+
+    def __init__(
+        self,
+        encoder: nn.Module,
+        decoder: nn.Module,
+    ) -> None:
+        """Initializes the VariationalAE class.
+
+        Args:
+            model_type (AutoEncoderType): The model type should be VARIATIONAL_AE.
```

Comment:
We're no longer accepting `model_type` in the init. So I think this is a leftover comment from a previous iteration.